protocols/server/cloud-events/main.tsp

```diff
@@ -0,0 +1,271 @@
+ */
+model MqttConnectEventFailureResponseProperties {
+    /**
+     * The failure code. It will be sent to the clients in the CONNACK packet as a return code (MQTT 3.1.1) or reason code (MQTT 5.0). Upstream webhook should select a valid integer value defined the MQTT protocols according to the protocol versions of the clients. If Upstream webhook sets an invalid value, clients will receive "unspecified error" in the CONNACK packet.
+     */
+    code: integer;
+
+    /**
+     * The reason for the failure. It's a human readable failure reason string designed for diagnostics. It will be sent to those clients whose protocols support reason string in the CONNACK packet. Now only MQTT 5.0 supports it.
+     */
+    reason?: string;
+
+    /**
+     * The user properties associated with the failure. They'll be converted to user properties in the CONNACK packet, and sent to clients whose protocols support user properties. Now only MQTT 5.0 supports user properties. Upstream webhook can use the property for additional diagnostic or other information.
+     */
+    userProperties?: MqttUserProperty[];
+}
+
+model CommonRequestHeaders {
+    /**
+     * The user the connection authed.
+     */
+    @header("ce-userId") userId: string;
+
+    /**
+     * The hub the connection belongs to.
+     */
+    @header("ce-hub") hub: string;
+
+    /**
+     * The connectionId is unique for the client connection.
+     */
+    @header("ce-connectionId") connectionId: string;
+
+    /**
+     * The name of the event without prefix.
+     */
+    @header("ce-eventName") eventName: string;
+
+    /**
+     * The subprotocol the client is using if any.
+     */
+    @header("ce-subprotocol") subprotocol: string;
+
+    /**
+     * Defines the state for the connection. You can use the same response header to reset the value of the state. Multiple `connectionState` headers aren't allowed. Do base64 encode the string value if it contains complex characters inside, for example, `base64(jsonString)` to pass complex object using this attribute.
+     */
+    @header("ce-connectionState") connectionState: string;
+
+    /**
+     * The signature for the upstream webhook to validate if the incoming request is from the expected origin. The service calculates the value using both primary access key and secondary access key as the `HMAC` key: `Hex_encoded(HMAC_SHA256(accessKey, connectionId))`. The upstream should check if the request is valid before processing it.
+     */
+    @header("ce-signature") signature: string;
+}
+
+/**
+ * Represents a request for a disconnection event.
+ */
+model DisconnectEventRequest {
+    reason: string;
+}
+
+/**
+ * Represents a request for an MQTT disconnection event.
+ */
+model MqttDisconnectedEventRequest extends DisconnectEventRequest {
+    mqtt: DisconnectedEventMqttProperties;
+}
+
+/**
+ * Represents the properties of an MQTT disconnection event.
+ */
+model DisconnectedEventMqttProperties {
```

Comment:
follow the same naming pattern as others to put Mqtt as the prefix, `MqttDisconnectedEventProperties`?